dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/jsl/impl/JobModelHandler.java

```diff
@@ -0,0 +1,179 @@
+public class JobModelHandler extends DefaultHandler {
+
+    private final static String sourceClass = JobModelHandler.class.getName();
+    private final static Logger logger = Logger.getLogger(sourceClass);
+
+    private static final String JCP_NAMESPACE_URI = "http://xmlns.jcp.org/xml/ns/javaee";
+    private static final String JAKARTA_NAMESPACE_URI = "https://jakarta.ee/xml/ns/jakartaee";
+
+    public final static String SCHEMA_LOCATION_V1 = "jobXML_1_0.xsd";
+
+    public final static String SCHEMA_LOCATION_V2 = "jobXML_2_0.xsd";
+
+    /**
+     * The JAXB unmarshaller that has been selected for the job XML file.
+     * This field will be null until the root job element has been
+     * parsed and its version has been determined.
+     */
+    UnmarshallerHandler ivHandler;
+
+    /**
+     * The buffered locator that will be set on ivHandler once created.
+     */
+    private Locator ivLocator;
+
+    /**
+     * The buffered startPrefixMapping events that will be replaced to ivHandler
+     * once created.
+     */
+    private final List<PrefixMapping> ivPrefixMappings = new ArrayList<PrefixMapping>();
+
+    final JSLValidationEventHandler validationHandler = new JSLValidationEventHandler();
+
+    @Override
+    public void setDocumentLocator(Locator locator) {
+        ivLocator = locator;
+    }
+
+    @Override
+    public void endDocument() throws SAXException {
+        if (ivHandler != null) {
+            ivHandler.endDocument();
+        }
+    }
+
+    @Override
+    public void startPrefixMapping(String prefix, String uri) throws SAXException {
+        // Buffer events until startElement is called.
+        if (ivHandler != null) {
+            ivHandler.startPrefixMapping(prefix, uri);
+        } else {
+            ivPrefixMappings.add(new PrefixMapping(prefix, uri));
+        }
+    }
+
+    @Override
+    public void endPrefixMapping(String prefix) throws SAXException {
+        if (ivHandler != null) {
+            ivHandler.endPrefixMapping(prefix);
+        }
+    }
+
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes atts) throws SAXException {
+        if (ivHandler == null) {
+            if (!"job".equals(localName) ||
+                (!(JCP_NAMESPACE_URI.equals(uri))
+                 && !(JAKARTA_NAMESPACE_URI.equals(uri)))) {
+                throw new SAXParseException("unexpected root element {" +
+                                            uri + "}" + localName, ivLocator);
+            }
+
+            String version = atts.getValue("", "version");
+
+            final String jaxbPackageName;
+            String schemaName = SCHEMA_LOCATION_V1;
+
+            if ("1.0".equals(version)) {
+                jaxbPackageName = "com.ibm.jbatch.jsl.model.v1";
+                schemaName = SCHEMA_LOCATION_V1;
+            } else {
+                jaxbPackageName = "com.ibm.jbatch.jsl.model.v2";
+                schemaName = SCHEMA_LOCATION_V2;
+            }
```

Comment:
So anything not explicitly 1.0 is treated as 2.0? I'm curious what the current behavior is.. does this version attribute not exist, not get evaluated (everything is assumed to be 1.0), or do we actually do something for invalid values?

Reply:
version is a required attribute so this isn't an issue.